gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/AbstractResource.java

```diff
@@ -125,6 +127,34 @@ protected boolean canReadAPIConfiguration() {
         return Stream.concat(streamUserMembership, streamGroupMembership);
     }
 
+    protected boolean canManageApi(final ApiEntity api) {
+        return isAdmin() || isDirectMember(api) || isMemberThroughGroup(api);
+    }
+
+    private boolean isDirectMember(ApiEntity api) {
+        return membershipService
+            .getMembershipsByMemberAndReference(USER, getAuthenticatedUser(), API)
+            .stream()
+            .anyMatch(membership -> membership.getReferenceId().equals(api.getId()));
+    }
+
+    private boolean isMemberThroughGroup(ApiEntity api) {
+        if (CollectionUtils.isEmpty(api.getGroups())) {
+            return false;
+        }
+
+        Set<String> groups = membershipService
+            .getMembershipsByMemberAndReference(USER, getAuthenticatedUser(), GROUP)
+            .stream()
+            .filter(m -> m.getRoleId() != null && roleService.findById(m.getRoleId()).getScope().equals(RoleScope.API))
+            .map(MembershipEntity::getReferenceId)
+            .collect(Collectors.toSet());
+
+        groups.retainAll(api.getGroups());
```

Comment:
It will be better to filter by `api.getgroups` before to call roleService to prevent useless database calls

Reply:
Ok sorry, I read the first commit